packages/admin-next/dashboard/src/routes/api-key-management/api-key-management-list/components/api-key-management-list-table/use-api-key-management-table-columns.tsx

```diff
@@ -0,0 +1,158 @@
+
+import { ActionMenu } from "../../../../../components/common/action-menu"
+import { DateCell } from "../../../../../components/table/table-cells/common/date-cell"
+import { StatusCell } from "../../../../../components/table/table-cells/common/status-cell"
+
+const columnHelper = createColumnHelper<PublishableApiKey>()
+
+export const useApiKeyManagementTableColumns = () => {
+  const { t } = useTranslation()
+
+  return useMemo(
+    () => [
+      columnHelper.accessor("title", {
+        header: t("fields.title"),
+        cell: ({ getValue }) => (
+          <div className="flex size-full items-center">
+            <span className="truncate">{getValue()}</span>
+          </div>
+        ),
+      }),
+      columnHelper.accessor("id", {
+        header: "Key",
+        cell: ({ getValue }) => {
+          const token = getValue()
+
+          return (
+            <div
+              className="bg-ui-bg-subtle border-ui-border-base box-border flex w-fit max-w-[220px] cursor-default items-center gap-x-0.5 overflow-hidden rounded-full border pl-2 pr-1"
+              onClick={(e) => e.stopPropagation()}
+            >
+              <Text size="xsmall" leading="compact" className="truncate">
+                {token}
+              </Text>
+              <Copy
+                content={token}
+                variant="mini"
+                className="text-ui-fg-subtle"
+              />
+            </div>
+          )
+        },
+      }),
+      columnHelper.accessor("revoked_at", {
+        header: t("fields.status"),
+        cell: ({ getValue }) => {
+          const revokedAt = getValue()
+
+          return (
+            <StatusCell color={revokedAt ? "red" : "green"}>
+              {revokedAt ? t("general.revoked") : t("general.active")}
+            </StatusCell>
+          )
+        },
+      }),
+      columnHelper.accessor("created_at", {
+        header: t("fields.created"),
+        cell: ({ getValue }) => {
+          const date = getValue()
+
+          return <DateCell date={date} />
+        },
+      }),
+      columnHelper.display({
+        id: "actions",
+        cell: ({ row }) => {
+          return <ApiKeyActions apiKey={row.original} />
+        },
+      }),
+    ],
+    [t]
+  )
+}
+
+const ApiKeyActions = ({ apiKey }: { apiKey: PublishableApiKey }) => {
+  const { mutateAsync: revokeAsync } = useAdminRevokePublishableApiKey(
+    apiKey.id
+  )
+  const { mutateAsync: deleteAsync } = useAdminDeletePublishableApiKey(
+    apiKey.id
+  )
+
+  const { t } = useTranslation()
+  const prompt = usePrompt()
+
+  const handleDelete = async () => {
+    const res = await prompt({
+      title: t("general.areYouSure"),
+      description: t("apiKeyManagement.deleteKeyWarning", {
+        title: apiKey.title,
+      }),
+      confirmText: t("actions.delete"),
+      cancelText: t("actions.cancel"),
+    })
+
+    if (!res) {
+      return
+    }
+
+    await deleteAsync()
+  }
```

Comment:
question: What should happen on delete? I am shown a 404. Should we navigate to `/api-management`?

![CleanShot 2024-02-23 at 16 30 34](https://github.com/medusajs/medusa/assets/59018053/a216d9b3-971e-4ebb-8906-9ebf90a7eec9)

Reply:
@kasperkristensen, just pinging in case you missed this one :)